main.go

```diff
@@ -150,27 +195,184 @@ func readConfig(filename string) (*config, error) {
+	// if no user then add one
+	if ok := errors.As(err, &pathError); ok {
+
+		cmd := exec.Command("usr/sbin/useradd", "-m", "-s", "/bin/bash", "-c", app, username)
+		stdoutStderr, err := cmd.CombinedOutput()
+		if err != nil {
+			log.Printf("user already exists: %s ", err.Error())
+		}
+
+		log.Printf("%s", stdoutStderr)
+	} else {
+		log.Printf("user already exists: %s skip create", username)
+	}
+}
+
+// getLastLogin for a user
+func getLastLogin(username string) time.Time {
+
+	cmd := exec.Command("last", "-1", username)
+	stdoutStderr, err := cmd.CombinedOutput()
+	if err != nil {
+		log.Printf("No lastlogin: %s ", err.Error())
+	}
+
+	s := string(stdoutStderr)
+	split := strings.Split(s, " ")
+	x := strings.TrimSpace(split[2] + " " + split[3] + " " + split[4] + " " + split[5] + " " + split[6] + " " + split[7])
+	return parseloginTime(x)
+}
+
+// parseloginTime get time from pased last command
+func parseloginTime(currenttime string) time.Time {
+
+	layout := time.ANSIC
+	t, err := time.Parse(layout, currenttime)
+
+	if err != nil {
+		fmt.Println(err)
+	}
+	return t
+}
+
+// createUser this create user is not exsits
+func createGroup(role string, username string) {
+	_, err := user.LookupGroup(role)
+	var pathError *user.NoFoundError
+	// if no group then add one
+	if ok := errors.As(err, &pathError); ok {
+
+		gid, err := user.AddGroup(role, username)
+
+		if err != nil {
+			log.Printf("cannot create group :%s ", err.Error())
+		}
+
+		log.Print("group: " + role + " created gid: " + strconv.Itoa(gid))
+
+	} else {
+		log.Printf("group already exists: %s skip set member", username)
+	}
+}
+
+// addUserToGroup add user to group by roles
+func addUserToGroup(role string, username string) {
+	_, err := user.LookupGroup(role)
+	var pathError *user.NoFoundError
+	// if no group then add one
+	if ok := errors.As(err, &pathError); ok {
+
+		err := user.AddUsersToGroup(role, username)
+
+		if err != nil {
+			log.Printf("cannot create membership :%s ", err.Error())
+		}
+		log.Printf("user added to :%s ", role)
+	}
+}
+
+// getAllUsers list all users from passwd
+func getAllUsers() []string {
+
+	var Users []string
+	file, err := os.Open("/etc/passwd")
+
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	defer file.Close()
+
+	reader := bufio.NewReader(file)
+
+	for {
+		line, err := reader.ReadString('\n')
+
+		// skip all line starting with #
+		if equal := strings.Index(line, "#"); equal < 0 {
+			// get the username and description
+			lineSlice := strings.FieldsFunc(line, func(divide rune) bool {
+				return divide == ':' // we divide at colon
+			})
+
+			if len(lineSlice) > 0 {
+				Users = append(Users, lineSlice[0])
+			}
+
+		}
+
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			fmt.Println(err)
+		}
+
+	}
+	return Users
+}
+
+// deleteOldUsers from added by pam modul
+func deleteOldUser(c config) {
+
+	for _, u := range getAllUsers() {
+
+		currentuser, err := user.LookupUser(u)
+		if err != nil {
```

Comment:
will panic with nilpointer if a error occurs, please continue the loop here

Reply:
i added continue to this loop